app/models/template.rb

```diff
@@ -114,7 +118,7 @@ def self.parse_metadata(text)
   #   :force - set to true if you want to bypass locked templates
   #   :associate - either 'new', 'always' or 'never', determines when the template should associate objects based on metadata
   #   :lock - lock imported templates (false by default)
-  def self.import!(name, text, options = {})
+  def self.import!(name:, text:, options: {})
```

Comment:
out of curiosity, why this is needed? the method `import_attrs_for` could have returned array and params could be passed with splat operator `*`

while I'm not against the change, I'm just curious why we change the method prototype

Reply:
I reverted this back, changed `import_params_for` to return an array.

The change was not strictly needed, the suggested solution with array and splat works. I could also do:

```ruby
def import_params_for(resource_name)
# ...
{ :name => name, :template => template, :options => options }
end

# then in controller

def import
     import_params = import_attrs_for(:provisioning_template)
    @provisioning_template = ProvisioningTemplate.import!(import_params[:name], import_params[:template], import_params[:options])
    process_response @provisioning_template
end

```

My reasoning was that when I need multiple different values returned from a method, hash is better option than array, because I do not have to keep track of order and I do not use array as if it were a tuple. And when I have proper keys in hash, why not use named params for destructuring instead of doing it manually.

Reply:
I see, thanks, I think it's good refactoring in general, but I'd prefer to keep the change minimal for now as we're getting closer to 1.18.